### PR TITLE
log mysql server version

### PR DESCRIPTION
**Pull request by razzeee**
Some work cherrypicked from @koying 

@MilhouseVH might like the mysql one

Thanks @koying

### CODE REVIEW

**Comment by koying** [comment]
You also need  https://github.com/koying/SPMC/commit/066c011f13853b20141dfee6367f294430a68f9b for the "system" screensaver thingy, on which the fixup applies ;)

**Comment by mazeyx** [comment]
plz cherrypick this for android omg, it fixed Zoom first time since 16.0 (since surface got enabled)

https://github.com/koying/SPMC/commit/845000d69def199761850dd7c990d02b359913ca

**Comment by Razzeee** [comment]
@mazeyx I would like approvement from @fritsch and/or @FernetMenta for that commit as that's their territory

@koying thanks, will pick it now

**Comment by mazeyx** [comment]
omg, this is the 1 issue its been driving me crazy since 16.0 alpha 4, let me explain, with mediacodec=surface on 4:3 SD content zoom on a bravia (android tv) wont stretch beyond the screensize causing the image to be squashed and really unwatchable, i opened multiple tickets and nothing has happend, on spmc it works perfectly, basically im not a dev but by looking at the code i guess it disables it for sd content which is great (no more switching it on/off all the time for me) i really need it on for hd content as its much faster and clearer, so i been switching it like a mad man all the time this would be a +100000000000 for me i really need it :p plz plz plz! @fritsch @FernetMenta

**Comment by fritsch** [comment]
@Razzeee I am the wrong for the ack. The special casing in Application.cpp is meehh as it's hard to follow, but the whole file is meh. 

For the second commit: If one likes statics, one can use statics, if one prefers members .. not on me to decide.

**Comment by fritsch** [comment]
Ah for: https://github.com/koying/SPMC/commit/845000d69def199761850dd7c990d02b359913ca#diff-1055cb645f9fd97653eafec81282c80eR303

This can easily go into MediaCodec itself, as it gets the hints and can decide itself, no need for the factory. On the other side - the others codecs also do their "dispatching" in the factory ...

**Comment by koying** [comment]
Re the 2nd one, it needs to be static or the log will be spammed with the line for every single new connections (and there's a bunch during the course of a Kodi session ;) ).

**Comment by fritsch** [comment]
@koying that's why I said: some want it static, others want to store that they have already logged in a member of the class itself.

**Comment by koying** [comment]
> This can easily go into MediaCodec itself

Nope. We need to call the codec twice, which can only be done in factory.
Surface or non-surface are functionally 2 codecs. It's just that there is so much common code that I prefer it that way, with a ctor param.

**Comment by koying** [comment]
> that's why I said: some want it static, others want to store that they have already logged in a member of the class itself.

Well, you can have a static member, if that's what you mean, but it has to be static anyway ;)

**Comment by mazeyx** [comment]
whatever fixes it ill be happy with, sorry i cant understand the lingo im not the only person who has this issue its been reported all over the kodi android bravia forum.

**Comment by fritsch** [comment]
@mazeyx non development related comments go into the forum, thx.

**Comment by mazeyx** [comment]
we tried that, we also tried the trac and sorry nothing happens :( if its at all possible please help us.

**Comment by fritsch** [comment]
@koying I don't understand the reasoning behind that as the Codec itself has a dependency to CSettings anyways and could decide by its own and depending on the settings in an encapsulated way, but the ```#ifdeffery is already there ...``` so if it helps and you want it that way ... I won't block that.

At the end it's a minor that would reduce the specialities of the way mediacodec is instantiated from the factory to make it easier for the future.

If this is also broken in Jarvis - we that one should be backported as we have responsibility for our users and Jarvis is not developed anymore anyways.

**Comment by fritsch** [comment]
@mazeyx I heard you, I heard you loud, will sing a song with you and shout: I with it for user`s purpose may it be wise and prosper - blessed be mediacodec's surface!

**Comment by mazeyx** [comment]
@fritsch :)

**Comment by fritsch** [comment]
@mazeyx in german english it even rhymes ...

**Comment by koying** [comment]
> I don't understand the reasoning behind that as the Codec itself has a dependency to CSettings anyways and could decide by its own and depending on the settings in an encapsulated way

As I said, surface and non-surface are 2 separate codecs for me, so I treat them as such.
But hey, I quit to avoid the bikeshedding on my PR's, not gonna start on someone else's ;)

**Comment by Razzeee** [comment]
Don't see any bikeshedding here, just discussion about technicality, which is fine for me. Fritsch even stated he won't block it.
2nd opinion from @FernetMenta would be highly appreciated.

**Comment by koying** [comment]
@Razzeee We are discussing a commit which is not even in you PR. That might qualify as bikeshedding :)

**Comment by fritsch** [comment]
@koying no need to be destructive I only wanted to help the user @mazeyx who joined this PR ... if you don't like that we cherry-pick code from your spmc fork, then clearly state this and it won't happen.

**Comment by Razzeee** [comment]
Yeah, but as this is spmc cherrypicks it is somewhat relevant. As I want to do more cherrypicks it's not really a problem, just wanted to get some of the smaller ones in before I go for the others.

**Comment by mazeyx** [comment]
@fritsch i really appreciate it, this has been affecting me greatly since jarvis alpha 4 (when mediacodec (surface) was added, i use the nightly now (as its more chance it might get fixed in some form) this is the first glimmer of hope this is why im so passionate about it.

**Comment by koying** [comment]
No issue whatsoever to cherry-pick from SPMC
I very much encourage you to do so in the contrary :)

Just don't ping me for endless discussions on code beauty, please.
Feel free to re-arrange the code the way you want...

**Comment by FernetMenta** [comment]
I don't get the sense of the screensaver commit.

**Comment by da-anda** [comment]
@FernetMenta I think the point here is to give users the option to either use the system screensaver (and power management) or one from Kodi. As of now, "dim" and "black" screensavers allowed the system screensaver to kick in (on Android at least) and users didn't like that. That's at least what I assume.

**Comment by FernetMenta** [comment]
I see two issues with this:

1) Android only in common code
2) Abuse of addon system

**Comment by da-anda** [comment]
I'd actually prefer if this would work on all platforms, and using a screensaver add-on to specify this behavior is IMO the cleanest and consistent way to this from a users perspective.

**Comment by FernetMenta** [comment]
> and using a screensaver add-on to specify this behavior is IMO the cleanest and consistent way to this from a users perspective

I disagree. From a user pov I am just wondering about the addons "Black" and "Dim" in the addon section. It shows me "auto-update" for both. Why pretending that those are addons? I don't see any advantage in this. In oder to be selectable with the same control those don't need to be addons.
From a code pov it is even worse.

**Comment by MilhouseVH** [comment]
+1 on the MySQL commit. :)

Won't be able to test it until tomorrow.

**Comment by Razzeee** [comment]
I agree with fernet, not to happy with that. I will drop that commit. 

The real reason I pinged you, was to know if this commit violates your vision or if it would be okay to cherrypick it. koying/SPMC@845000d
Haven't looked closely yet, but will try to also check out later.

**Comment by FernetMenta** [comment]
@Razzeee the linked commit in principle is ok but it won't apply because the code has changed. You need to do this change directly in CDVDVideoCodecAndroidMediaCodec::Open

**Comment by Razzeee** [comment]
Thank you fernet, will have a try at doing that.

I dropped the screensaver commit. And this should be fine, when @MilhouseVH gives his okay.

**Comment by mazeyx** [comment]
thanks @Razzeee if you are able i`ll be forever grateful, this has been a big issue for me, i`m happy to help test it out. :)

**Comment by MilhouseVH** [comment]
Looks good:
```
20:55:10 T:140345222879360    INFO: MYSQL: Connected to version 5.5.21
```
There doesn't appear to be a way to confirm if the server is actually MySQL, but I guess we can infer from the version number whether the server MySQL or something else (ie. MariaDB).

**Comment by Razzeee** [comment]
Looking at the mysql documentation doesn't offer a solution that shows the product name. So I guess we're out of luck.

**Comment by MrMC** [comment]
lol, will someone hit the green button and put this PR out of limbo ?

**Comment by Razzeee** [comment]
jenkins build and merge
